core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala

```diff
@@ -213,7 +213,7 @@ class TopicCommandWithAdminClientTest extends KafkaServerTestHarness with Loggin
   def testCreateWithNegativeReplicationFactor(): Unit = {
     intercept[ExecutionException] {
       topicService.createTopic(new TopicCommandOptions(
-        Array("--partitions", "2", "--replication-factor", "-1", "--topic", testTopicName)))
+        Array("--partitions", "2", "--replication-factor", "-2", "--topic", testTopicName)))
```

Comment:
Do we have to allow the user to specify `--replication-factor -1`? An alternative would be to continue to treat this as invalid input and make the `--replication-factor` argument optional. Similarly for `--partitions`.

Reply:
I made this the case and I also changed it to reflect the NewTopic API where you must specify both partitions and replicas or neither. We can easily change this in the future to allow specifying one or the other.